From 28b41f521c5c22c329f499379e7ee066a9472d23 Mon Sep 17 00:00:00 2001
From: Han Xu <b45815@freescale.com>
Date: Fri, 8 May 2015 11:00:18 -0500
Subject: [PATCH 0740/5242] MLK-10868: ARM:dts: support 4 qspi chips on
 imx7d-12x12-lpddr3-arm2

commit  4c1364f2a445c017da0d0ef0e2d0d7b8a5b25bc4 from
https://source.codeaurora.org/external/imx/linux-imx.git

enabled 4 qspi chips on imx7d-12x12-lpddr3-arm2 board in dts

Signed-off-by: Han Xu <b45815@freescale.com>
(cherry picked from commit a4e10bf5e94d883c06f41b2c61f80ce23946612b)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-qspi.dts |   17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-qspi.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-qspi.dts
index db0133b..4f2eb33 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-qspi.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-qspi.dts
@@ -42,6 +42,7 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_qspi1_1>;
 	status = "okay";
+	fsl,qspi-has-second-chip = <1>;
 	ddrsmp=<0>;
 
 	flash0: n25q256a@0 {
@@ -59,4 +60,20 @@
 		spi-max-frequency = <29000000>;
 		reg = <1>;
 	};
+
+	flash2: n25q256a@2 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "micron,n25q256a";
+		spi-max-frequency = <29000000>;
+		reg = <2>;
+	};
+
+	flash3: n25q256a@3 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "micron,n25q256a";
+		spi-max-frequency = <29000000>;
+		reg = <3>;
+	};
 };
-- 
1.7.9.5

